### PR TITLE
network: add warning for kickstart network configuration when running…

### PR DESCRIPTION
**Pull request by rvykydal**
… from nfs

Resolves: RHEL-35250

Adds warning to tty1 and a log message.
Screenshot from 2024-10-22 16-06-16

### CODE REVIEW

**Comment by rvykydal** [comment]
Is it possible that the installation freezes before the warning is issued? Which is only after the kickstart network configuration starts (asynchronously) so that we know if there are some devices configured.

**Comment by rvykydal** [comment]
> Is it possible that the installation freezes before the warning is issued? Which is only after the kickstart network configuration starts in `Apply kickstart` (asynchronously) so that we know if there are some devices configured.
> 

Updated to use `Kickstarted` API instead.
```
10:54:58,146 DBG payload.dnf.payload: Use the DNF source from the Anaconda configuration file.
10:54:58,200 DBG localization: setting locale to: en_US.UTF-8
10:54:58,219 DBG network: Initialization started.
10:54:58,230 DBG network: Devices found: ['enp0s22']
10:54:58,233 WRN network: Using kickstart network configuration with installer image (stage2) provided via nfs server can freeze the installation.
10:54:58,240 DBG network: Running task Apply kickstart
10:54:58,400 DBG network: Apply kickstart result: ['enp0s22']
10:54:58,467 DBG network: Running task Dump missing config files
10:54:58,613 DBG network: Dump missing config files result: []
10:54:58,625 DBG network: Creating network configurations.
10:54:58,651 DBG network: Initialization finished.
```

File: pyanaconda/core/util.py [hunk 1]
```diff
@@ -911,3 +911,13 @@ def get_image_packages_info(max_string_chars=0):
...
                 info_lines.append(' '.join(line.strip() for line in lines))
     return info_lines
+
+
+def is_stage2_on_nfs():
+    """Is the installation running from image mounted via NFS?"""
+    for line in open("/proc/mounts").readlines():
+        values = line.split()
+        if len(values) > 2:
+            if values[1] == DRACUT_REPO_DIR and values[2] == "nfs":
```
[the remark this answers is not in the document]
Comment (rvykydal) [reply pyanaconda/core/util.py:921]: Nice catch, thank you!

File: pyanaconda/network.py [hunk 2]
```diff
@@ -258,6 +258,12 @@ def initialize_network():
     log.debug("Devices found: %s",
               [dev.device_name for dev in get_supported_devices()])
 
+    if util.is_stage2_on_nfs() and network_proxy.Kickstarted:
+        msg = "Using kickstart network configuration with installer image (stage2) provided " \
+            "via nfs server can freeze the installation."
+        log.warning(msg)
+        print("WARNING:", msg)
```
Comment (KKoukiou) [inline pyanaconda/network.py:265]: Is this forgetten debug log? Why we need log.warning and print?

File: tests/unit_tests/pyanaconda_tests/core/test_util.py [hunk 3]
```diff
@@ -815,6 +817,134 @@ def test_dracut_eject(self):
                     file_contents = "\n".join(f.readlines())
                     assert "eject " + devname in file_contents
 
+    @patch("pyanaconda.core.util.open")
+    def test_is_stage2_on_nfs(self, mock_open):
+        """Test check for installation running on nfs."""
+        nfs_source_mounts = """
```
Comment (KKoukiou) [inline tests/unit_tests/pyanaconda_tests/core/test_util.py:823]: It's huge text strings. I will not insist, but I would suggest to put the mounts in a separate file, which we can include here.

Comment (rvykydal) [reply tests/unit_tests/pyanaconda_tests/core/test_util.py:823]: It is, I wanted to have real output there but I think I'll rather trim them to a few lines?

File: pyanaconda/network.py (hunk 2 above)
Comment (rvykydal) [reply pyanaconda/network.py:265]: I think we want both print - to show the warning in the UI (on the tty1), and log - to have the message in the logs. The goal is to make it discoverable when hitting the situation of the freeze.

I was playing also with `anaconda.loggers.get_stdout_logger` that goes both on the screen and into anaconda.log but it did not look as I wanted (warning enough) on the tty1.
Screenshot from 2024-10-24 11-03-01

**Comment by jkonecny12** [comment]
Seems that you have one commit extra here which shouldn't be here.

**Review by jkonecny12 — APPROVED** [review]

**Review by jstodola — APPROVED** [review]

File: tests/unit_tests/pyanaconda_tests/core/test_util.py (hunk 3 above)
Comment (rvykydal) [reply tests/unit_tests/pyanaconda_tests/core/test_util.py:823]: Picked only a few lines.

**Pull request merged**